### PR TITLE
Blog-post: Bazel - a multilanguage build system

### PR DESCRIPTION
**Pull request by laurentlb**
Draft was reviewed by @dslomov, David, and @spomorski.

### CODE REVIEW

File: _posts/2018-12-05-multilanguage-build-system.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - laurentlb
+---
+
+Part of [Bazel vision](https://docs.bazel.build/versions/master/bazel-overview.html)
+is to create a multilanguage build system: it should be possible to seamlessly
+use and mix multiple languages in a codebase and to orchestrate the builds and
+the tests with a single build tool. As a codebase gets bigger, this feature
+becomes more and more important.
+
+When a user wants a build system to be multilanguage, they might mean:
```
Comment (meisterT) [inline _posts/2018-12-05-multilanguage-build-system.md]: nit: multi-language?

File: _posts/2018-12-05-multilanguage-build-system.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+layout: posts
+title: "Bazel - a multilanguage build system"
+authors:
+  - laurentlb
+---
+
+Part of [Bazel vision](https://docs.bazel.build/versions/master/bazel-overview.html)
+is to create a multilanguage build system: it should be possible to seamlessly
+use and mix multiple languages in a codebase and to orchestrate the builds and
```
Comment (spomorski) [inline _posts/2018-12-05-multilanguage-build-system.md]: "builds and tests" - remove "the" from before both words

File: _posts/2018-12-05-multilanguage-build-system.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+However, quantity doesn't imply quality. As with any open-source code published
+on the Internet, there is no guarantee that a repository is properly designed
+and maintained. To help users, we want to promote a set of high-quality rules
+that they can trust. Our website lists [a few external rules](
+https://docs.bazel.build/versions/master/be/overview.html#additional-rules), but
+we are working on a more principled process for recommendations - for example,
+we should recommend only rules that we have properly tested.
+
+It's hard to find an objective measure of the quality, but we can look at the
```
Comment (spomorski) [inline _posts/2018-12-05-multilanguage-build-system.md]: "measure of quality" - delete "the"

File: _posts/2018-12-05-multilanguage-build-system.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+*   Is the build system extensible? In other words, can I add support for a new
+    language, even if it doesn't have much adoption yet, or if it is internal?
+*   How many languages are already supported?
+*   How big is the ecosystem?
+*   Does it provide a high level of quality for the languages I use?
+*   Can I use it in production?
+
+Bazel allows you to [create your own rules](https://docs.bazel.build/versions/master/skylark/concepts.html).
+More importantly, however, it allows rules to communicate and pass information
+using "providers." For example, if a rule depends on a C++ library, it might
```
Comment (meisterT) [inline _posts/2018-12-05-multilanguage-build-system.md]: nit: switch ." to ".

Comment (laurentlb) [reply _posts/2018-12-05-multilanguage-build-system.md]: @spomorski didn't like it.
It's common in the US to put the punctuation inside quotes (but I'm not a fan of this convention).

File: _posts/2018-12-05-multilanguage-build-system.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+page lists Bazel rules for many languages and tools. If you use a well-known
+language, it's likely you'll find an existing rule for it.
+
+However, quantity doesn't imply quality. As with any open-source code published
+on the Internet, there is no guarantee that a repository is properly designed
+and maintained. To help users, we want to promote a set of high-quality rules
+that they can trust. Our website lists [a few external rules](
+https://docs.bazel.build/versions/master/be/overview.html#additional-rules), but
+we are working on a more principled process for recommendations - for example,
+we should recommend only rules that we have properly tested.
```
Comment (meisterT) [inline _posts/2018-12-05-multilanguage-build-system.md]: Didn't you have a doc about how a good rule set looks like (I can't find it just now). If you have it, please link it here.

Comment (laurentlb) [reply _posts/2018-12-05-multilanguage-build-system.md]: This? https://docs.bazel.build/versions/master/bazel-vision.html#so-what-is-a-good-ruleset

There's already a link to the vision.

Comment (meisterT) [reply _posts/2018-12-05-multilanguage-build-system.md]: ah, forgot that this is part of the vision doc. If you feel like it, then link to the specific subsection from here, but I can also live without

**Pull request merged**